plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleEngineOrchestratorPlugin.java

```diff
@@ -67,7 +72,71 @@ public List<AbstractCredentialInfo> getCredentialDefinitions() {
     public List<Metric> getMetrics(AbstractCredentialInfo credential,
                                    DeployResource deployResource,
                                    MonitorResourceType monitorResourceType) {
-        return null;
+        List<Metric> metrics = new ArrayList<>();
+        String region = deployResource.getProperties().get("region");
+        if (StringUtils.isBlank(region)) {
+            log.error("GetMetrics param region is blank. resource id:{}",
+                    deployResource.getResourceId());
+            return metrics;
+        }
+        String queryProjectIdUrl =
+                FlexibleEngineMonitorConverter.buildProjectQueryUrl(region).toString();
+        FlexibleEngineMonitorClient monitorClient =
+                getFlexibleEngineMonitorClient((CredentialDefinition) credential);
+        String projectId = queryProjectId(monitorClient, queryProjectIdUrl);
```

Comment:
Even getting the projectID does not work via Huaweicloud SDKs? Does the APIs do not match or do we have some other issue?

Reply:
There is no API matched in the  Flexible engine SDK and  in the Huaweicloud SDK the endponit of service is fixed. In addition to using API queries, we can also add "PROJECT_ID" like 'AK' in the credential provided by user. But I think calling API is better. I will try to use Spring Resttemplate to call this API . Is it ok?